### PR TITLE
Redirect to non-www on API browser

### PR DESCRIPTION
**Pull request by andrebola**
issue #965

The method GenericAPIView.initial raise an exception if the user is not logged before setting the value of self.accepted_renderer, so it can't be evaluated if the user is accessing through the browser.

Therefore I'm overriding the method finalize_response, where the value of self.accepted_renderer indicates if it's using the browser.

### CODE REVIEW

**Comment by ffont** [comment]
This looks good to me, but because we don't have the redirect from `www` to `non-www`enabled now we can't test. Let's wait until we have it enabled again and then we can merge and test.

File: apiv2/apiv2_utils.py [hunk 1]
```diff
@@ -79,6 +81,22 @@ def initial(self, request, *args, **kwargs):
...
+        """ This method is overriden to make a redirect when the user is using the interactive API browser and
+        with 'www' sub-domain. The problem is that we can't check if it's accessing through the interactive browser
+        inside the 'initial' method because it raises an exception when the user is not logged in, that exception is
+        handled by 'finalize_response' method of APIView.
+        """
+        response = super(GenericAPIView, self).finalize_response(request, response, *args, **kwargs)
+
+        # If the user is using the interactive API browser and the www in the domain we redirect to nowww.
+        host = request.get_host()
+        if type(response.accepted_renderer) == type(BrowsableAPIRenderer()) and 'www' in host:
```
Comment (alastair) [inline apiv2/apiv2_utils.py]: `host.startswith('www.')`

File: apiv2/apiv2_utils.py [hunk 2]
```diff
@@ -79,6 +81,22 @@ def initial(self, request, *args, **kwargs):
...
+            domain = "https://%s" % Site.objects.get_current().domain
```
Comment (alastair) [inline apiv2/apiv2_utils.py]: let's use the scheme from `request` here - if it's http then we can rely on haproxy to redirect us again to https

by using the request scheme we can test locally more easily

File: apiv2/apiv2_utils.py (hunk 1 above)
Comment (alastair) [inline apiv2/apiv2_utils.py]: I prefer to use `isinstance` instead of `type`, because it works for subclasses too

**Comment by ffont** [comment]
I'm working on this PR, I'll make changes as @alastair requests and some other small ones. @andrebola I'll take it from here.

**Pull request merged**